rnaseq-cgl-pipeline/wrapper.py

```diff
@@ -1,102 +1,203 @@
+    work_dir = os.path.join(mount, 'Toil-RNAseq-' + str(uuid4()))
+    os.makedirs(work_dir)
+    log.info('Temporary directory created: {}'.format(work_dir))
+    config_path = os.path.join(work_dir, 'toil-rnaseq.config')
+    job_store = os.path.join(args.resume, 'jobStore') if args.resume else os.path.join(work_dir, 'jobStore')
+    with open(config_path, 'w') as f:
+        f.write(generate_config(args.star, args.rsem, args.kallisto, mount, args.save_bam, args.save_wiggle))
+    command = ['toil-rnaseq', 'run',
+               job_store,
+               '--config', config_path,
                '--workDir', work_dir,
-               '--star-index', 'file://' + args.star,
-               '--rsem-ref', 'file://' + args.rsem,
-               '--kallisto-index', 'file://' + args.kallisto,
-               '--sample-urls']
-    command.extend('file://' + x for x in args.samples)
-    if args.restart:
+               '--retryCount', '1']
+    if args.resume:
         command.append('--restart')
+    if args.cores:
+        command.append('--maxCores={}'.format(args.cores))
+    command.append('--samples')
+    command.extend('file://' + x for x in args.samples)
     try:
         subprocess.check_call(command)
+    except subprocess.CalledProcessError as e:
+        print(e.message, file=sys.stderr)
     finally:
+        log.info('Pipeline terminated, changing ownership of output files from root to user.')
         stat = os.stat(mount)
         subprocess.check_call(['chown', '-R', '{}:{}'.format(stat.st_uid, stat.st_gid), mount])
-        shutil.rmtree(work_dir)
+        if not args.no_clean:
+            log.info('Cleaning up temporary directory: {}'.format(work_dir))
+            shutil.rmtree(work_dir)
+        else:
+            log.info('Flag "--no-clean" was used, therefore {} was not deleted.'.format(work_dir))
+
+
+def generate_config(star_path, rsem_path, kallisto_path, output_dir, disable_cutadapt, save_bam, save_wiggle):
+    cutadapt = True if not disable_cutadapt else False
+    return textwrap.dedent("""
+        star-index: file://{star_path}
+        kallisto-index: file://{kallisto_path}
+        rsem-ref: file://{rsem_path}
+        output-dir: {output_dir}
+        cutadapt: {cutadapt}
+        fastqc: true
+        fwd-3pr-adapter: AGATCGGAAGAG
+        rev-3pr-adapter: AGATCGGAAGAG
+        ssec:
+        gtkey:
+        wiggle: {save_wiggle}
+        save-bam: {save_bam}
+        ci-test:
+    """[1:].format(star_path=star_path, rsem_path=rsem_path, kallisto_path=kallisto_path, output_dir=output_dir,
+                   cutadapt=cutadapt, save_wiggle=save_wiggle, save_bam=save_bam))
+
+
+class UserError(Exception):
+    pass
+
+
+def require(expression, message):
+    if not expression:
+        raise UserError('\n\n' + message + '\n\n')
+
+
+def check_for_input(tool_input, name):
+    require(tool_input, 'Cannot find {0} input, please use --{0} and provide full path to file.'.format(name))
```

Comment:
If a glob matches more than one thing, taking the first thing the glob matches isn't necessarily right. I would throw an exception if tool_input is not just a single item.